public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -283,7 +286,17 @@ function getLinkForLoki(span: TraceSpan, options: TraceToLogsOptions, dataSource
   return dataLink;
 }
 
-function getLinkForElasticsearch(
+// we do not have access to the dataquery type for opensearch,
+// so here is a minimal interface that handles both elasticsearch and opensearch.
+interface ElasticSearchOrOpensearchQuery extends DataQuery {
+  query: string;
+  metrics: Array<{
+    id: string;
+    type: 'logs';
+  }>;
+}
+
+function getLinkForElasticsearchOrOpensearch(
```

Comment:
nit: we should use either `ElasticSearch` or `Elasticsearch` :)

Reply:
good catch! i checked and "Elasticsearch" seems to be the right way to capitalize it, so i did it that way 👍